packages/gatsby-plugin-page-creator/src/is-valid-collection-path-implementation.ts

```diff
@@ -25,7 +25,7 @@ export function isValidCollectionPathImplementation(
 
     try {
       assert(opener, `{`, ``) // This is a noop because of the opening check, but here for posterity
-      assert(model, /^[A-Z][a-zA-Z]+$/, errorMessage(part))
+      assert(model, /^[a-zA-Z_][a-zA-Z0-9_]+$/, errorMessage(part))
```

Comment:
nit:
```suggestion
      assert(model, /^[\w_][\w\d_]+$/, errorMessage(part))
```

Assuming this is checking for ids, in JS `$` is also a valid character. Not sure if that applies here

Reply:
This is the same regex as `extractFieldGraphQLModel` and tries to enforce: https://spec.graphql.org/draft/#sec-Names